src/mono/mono/mini/mini-arm64.c

```diff
@@ -5924,6 +5924,11 @@ mono_arch_emit_prolog (MonoCompile *cfg)
 		code = emit_addx_imm (code, cfg->arch.args_reg, ARMREG_FP, cfg->stack_offset);
 	}
 
+	/* Call the init wrapper which checks if the methos needs to be initialised or not */
+	code = emit_imm (code, ARMREG_R0, cfg->method_index);
+	code = emit_call (cfg, code, MONO_PATCH_INFO_METHOD, NULL, mono_marshal_get_aot_init_wrapper (AOT_INIT_METHOD));
```

Comment:
so this is always a real `bl`. Because we're in the backend there is no chance to inline the init wrapper into the caller, right?

Can we emit a call to the init wrapper in method-to-ir in the first basic block of the method? Or does it really need to be done by the backend for some reason?

Reply:
The init calls need to be done as early as possible, doing it in the front end is possible, but it increases the change some code is emitted before it which depends on the initialization.